cmd/libsnap-confine-private/string-utils.c

```diff
@@ -72,6 +72,33 @@ bool sc_startswith(const char *str, const char *prefix)
 	return strncmp(str, prefix, xlen) == 0;
 }
 
+bool sc_is_beneath(const char *path, const char *parent_dir)
+{
+	if (!path || !parent_dir) {
+		return false;
+	}
+	// The path "path" is beneath the directory "parent_dir" if "path" is a
+	// prefix of "parent_dir" with the following additional restrictions.
```

Comment:
This comment isn't right. This is the API: `sc_startswith(const char *str, const char *prefix)` and you call `if (sc_startswith(path, parent_dir)) {` in the next line, so the comment should read:
```
// The path "path" is beneath the directory "parent_dir" if "parent_dir"
// is a prefix of "path" with the following additional restrictions.
```

Reply:
I've removed this function now. The commit message left a remark about unaddressed concerns in case anyone wants to bring it back.